pkg/manila/cronjob.go

```diff
@@ -0,0 +1,124 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manila
+
+import (
+	common "github.com/openstack-k8s-operators/lib-common/modules/common"
+	manilav1 "github.com/openstack-k8s-operators/manila-operator/api/v1beta1"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strconv"
+)
+
+const (
+	// DBPurgeCommand -
+	DBPurgeCommand = "/usr/bin/manila-manage --config-dir /etc/manila/manila.conf.d db purge "
+)
+
+// CronJob func
+func CronJob(
+	instance *manilav1.Manila,
+	labels map[string]string,
+	annotations map[string]string,
+) *batchv1.CronJob {
+	runAsUser := int64(0)
+	var config0644AccessMode int32 = 0644
+
+	args := []string{"-c"}
+	if instance.Spec.Debug.DBSync {
+		args = append(args, common.DebugCommand)
+	} else {
+		args = append(args, DBPurgeCommand+strconv.Itoa(DBPurgeAge))
+	}
+
+	parallelism := int32(1)
+	completions := int32(1)
+
+	cronJobVolume := []corev1.Volume{
+		{
+			Name: "db-purge-config-data",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					DefaultMode: &config0644AccessMode,
+					SecretName:  instance.Name + "-config-data",
+					Items: []corev1.KeyToPath{
+						{
+							Key:  DefaultsConfigFileName,
+							Path: DefaultsConfigFileName,
```

Comment:
possibly a dumb question, but I'm assuming this value (DefaultsConfigFileName) is defined elsewhere and we can just consume it, am I correct?

Reply:
yeah, it's a `const` defined in the same package and it represents the global config where the service can find the db connection string (which is the only thing we need here).
I only added this config to the cronJob (instead of the whole manila conifig) to limit the mount to what is just required.

Reply:
Ah, great! Thanks